### PR TITLE
Feat/hng 4 create signup page

### LINKED ISSUE #4: [FEAT] - Authentication Page -> Sign Up Default
### Description
Implement the Sign Up default page as shown in the design

### Acceptance Criteria
- Responsive across all screen sizes (desktop, tablet and mobile)
- Write tests for your code.
- Add validation to all fields as shown in the design. Refer to the Email/Error Page
- Use components where necessary (header, footer, buttons, input fields and so on)
- Implement a focus state on all input fields.
Refer to the Email/Filling page.

### Tasks 
- [x] Implementation of the sign up page.
- [x] Validation of all input fields showing the appropriate error messages 
- [x] This page should be responsive on all screen sizes 
- [x] Write unit test for your implementation.

### Timeframe
24 hours

[Link to Design](
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1-17895&t=p2gECcZQmYqvvrn7-4) 

### Preview
Sign up Default

### PR DESCRIPTION
**Pull request by omotayobolu**
# Description

**Closes #4**

# Changes proposed

## What were you told to do?

Create a sign up page

## What did you do?

I created a sign up page, with form validdation using zod.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

FireShot Capture 015 - Create Next App - localhost

### CODE REVIEW

File: public/bi_facebook.svg
Comment (skeby) [inline public/bi_facebook.svg:1]: Your icons should be in a public/signup/icons directory

**Review by skeby — CHANGES_REQUESTED** [review]
fix up

File: public/bi_facebook.svg
Comment (omotayobolu) [reply public/bi_facebook.svg:1]: alright, on it

**Comment by omotayobolu** [comment]
> fix up

it's been fixed

**Review by oreoluwa212 — APPROVED** [review]

**Review by skeby — APPROVED** [review]

**Pull request merged**